TestSolutions/AnalyzerViolations.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Lombiq.Analyzers.ProjectReference;
+
+// This file has intentional analyzer violations. Only edit the one in the TestSolutions directory. Any copy in the
+// subdirectories will be copied over during build. Copying is necessary so the source files are in a directory with an
+// existing parent .editorconfig file. (Linking will cause them to physically be outside of the .editorconfig's area of
+// effect.) Also please commit the updated version everywhere, because if the file is not present before the build
+// starts then MSBuild won't include it despite the <Copy> task. For this purpose the <Copy> task is only there to
```

Comment:
> if the file is not present before the build starts then MSBuild won't include it

I've had to work around this behavior in NE, too, and was able to get it to work by adding the generated files to the `EmbeddedResource` item group like [this](https://github.com/Lombiq/NodeJs-Extensions/blob/dev/Lombiq.NodeJs.Extensions/Lombiq.NodeJs.Extensions.targets#L124).

Reply:
Great, thanks Oliver, you are a true life saver. Because after this the _.editorconfig_ (which is copied by the _Lombiq.Analyzers/Build.props_) started to do just the same. (see [this](https://github.com/Lombiq/.NET-Analyzers/runs/7567993084?check_suite_focus=true); the test for "Local ProjectReference" passes because I committed the .editorconfig [here](https://github.com/Lombiq/.NET-Analyzers/pull/57/commits/1d4231838fb4d97170fc316e59f96a090645653b))

Reply:
Let's see!

Reply:
[Sadly it doesn't work.](https://github.com/Lombiq/.NET-Analyzers/runs/7568514194?check_suite_focus=true)

Reply:
Can't you copy `AnalyzerViolations.cs` at an even earlier stage? Like `BeforeBuild` e.g.?

Reply:
Actually, [even earlier](https://stackoverflow.com/a/51609999/177710): `PrepareForBuild`.

Reply:
Already tried it, no change. I'm currently using ResolveProjectReferences which is even earlier; it's the earliest possible target (I've checked it in the build logs).
Out of desperation I've even tried using Clean. Then if I do `dotnet clean; dotnet build` it works, but `dotnet build --no-incremental` (rebuild) doesn't.

Reply:
For me the order is the opposite:

![image](https://user-images.githubusercontent.com/164360/181640530-95193705-5573-46fb-8592-410ac6d9550e.png)

Reply:
Weird. But I won't hang up on it since it doesn't work either way